L1Trigger/L1TTrackMatch/plugins/L1TkElectronTrackProducer.cc

```diff
@@ -0,0 +1,340 @@
+
+// Matching Algorithm
+#include "L1Trigger/L1TTrackMatch/interface/L1TkElectronTrackMatchAlgo.h"
+#include "L1Trigger/L1TTrackMatch/interface/L1TkElectronEtComparator.h"
+#include "L1Trigger/L1TTrackMatch/interface/pTFrom2Stubs.h"
+
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/L1Trigger/interface/EGamma.h"
+
+#include <string>
+#include "TMath.h"
+
+using namespace l1t;
+
+//
+// class declaration
+//
+
+class L1TkElectronTrackProducer : public edm::EDProducer {
+public:
+  typedef TTTrack<Ref_Phase2TrackerDigi_> L1TTTrackType;
+  typedef std::vector<L1TTTrackType> L1TTTrackCollectionType;
+
+  explicit L1TkElectronTrackProducer(const edm::ParameterSet&);
+  ~L1TkElectronTrackProducer() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginJob() override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+
+  //virtual void beginRun(edm::Run&, edm::EventSetup const&);
+  //virtual void endRun(edm::Run&, edm::EventSetup const&);
+  //virtual void beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
+  //virtual void endLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
+
+  float isolation(const edm::Handle<L1TTTrackCollectionType>& trkHandle, int match_index);
+  double getPtScaledCut(double pt, std::vector<double>& parameters);
+  bool selectMatchedTrack(double& d_r, double& d_phi, double& d_eta, double& tk_pt, float& eg_eta);
+
+  // ----------member data ---------------------------
+  std::string label;
+
+  float ETmin;  // min ET in GeV of L1EG objects
+
+  float DRmin;
+  float DRmax;
+  float PTMINTRA;
+  float maxChi2IsoTracks;
+  unsigned int minNStubsIsoTracks;
+
+  bool PrimaryVtxConstrain;  // use the primary vertex (default = false)
+  float DeltaZ;              // | z_track - z_ref_track | < DeltaZ in cm.
+                             // Used only when PrimaryVtxConstrain = True.
+  float IsoCut;
+  bool RelativeIsolation;
+
+  float trkQualityChi2;
+  bool useTwoStubsPT;
+  float trkQualityPtMin;
+  std::vector<double> dPhiCutoff;
+  std::vector<double> dRCutoff;
+  std::vector<double> dEtaCutoff;
+  std::string matchType;
+
+  const edm::EDGetTokenT<EGammaBxCollection> egToken;
+  const edm::EDGetTokenT<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > > trackToken;
+};
+
+//
+// constructors and destructor
+//
+L1TkElectronTrackProducer::L1TkElectronTrackProducer(const edm::ParameterSet& iConfig)
+    : egToken(consumes<EGammaBxCollection>(iConfig.getParameter<edm::InputTag>("L1EGammaInputTag"))),
+      trackToken(consumes<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > >(
+          iConfig.getParameter<edm::InputTag>("L1TrackInputTag"))) {
+  label = iConfig.getParameter<std::string>("label");  // label of the collection produced
+                                                       // e.g. EG or IsoEG if all objects are kept
```

Comment:
put block comments before code line to avoid massive unnecessary whitespace

Reply:
done